lib/Doctrine/DBAL/DBALException.php

```diff
@@ -28,21 +35,57 @@ class DBALException extends Exception
      */
     public static function driverExceptionDuringQuery(Driver $driver, Throwable $driverEx, $sql, array $params = [])
     {
-        $msg = "An exception occurred while executing '" . $sql . "'";
-        if ($params) {
-            $msg .= ' with params ' . self::formatParameters($params);
-        }
-        $msg .= ":\n\n" . $driverEx->getMessage();
+        $messageFormat = <<<MESSAGE
+An exception occurred while executing "%s"%s:
+
+%s.
+MESSAGE;
+
+        $message = sprintf(
+            $messageFormat,
+            $sql,
+            $params !== [] ? sprintf(' with params %s', self::formatParameters($params)) : '',
+            $driverEx->getMessage()
+        );
 
-        return static::wrapException($driver, $driverEx, $msg);
+        return static::wrapException($driver, $driverEx, $message);
     }
 
     /**
      * @return self
      */
     public static function driverException(Driver $driver, Throwable $driverEx)
     {
-        return static::wrapException($driver, $driverEx, 'An exception occurred in driver: ' . $driverEx->getMessage());
+        return static::wrapException($driver, $driverEx, sprintf('An exception occurred in driver with message: %s', $driverEx->getMessage()));
+    }
+
+    public static function formatValue($value) : string
```

Comment:
Is there a better place for this? or do you want me to create a new class for it? New class probably makes the most sense.

Reply:
`Doctrine\DBAL\Exception\VarFormatter::__invoke()`?

Reply:
If we use `__invoke()`, would it be named like an action? e.g. `FormatVariable` instead of `VarFormatter`

Reply:
Yeah that might make more sense 👍